src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Span.cs

```diff
@@ -88,6 +88,7 @@ public static partial class JsonSerializer
                 ThrowHelper.ThrowArgumentNullException(nameof(jsonTypeInfo));
             }
 
+            jsonTypeInfo.EnsureConfigured();
```

Comment:
isn't it better to have this just in ReadFromSpan or rather closer to wherever it needs to be configured? (assuming it's not already there) - I'd avoid adding to in any "proxy" methods

Reply:
For source JsonTypeInfo sources like the `JsonSerializerOptions` cache it is already guaranteed for the value to be configured. Which is why we only need to do this in public methods that accept `JsonTypeInfo<T>`. I've replaced the `EnsureConfigured()` calls with asserts to make sure nothing slips.